src/vm/internal/otable.cc

```diff
@@ -1,271 +1,129 @@
+            j->second.erase(l);
+        }
+        else {
+            return false;
+        }
+        if( j->second.size() == 0 ) {
+            children_.erase(j);
+        }
+        return true;
+    }
+    else {
+        return false;
+    }
 }
 
-/*
- * Lookup an object in the hash table; if it isn't there, return null.
- * This is only different to find_object_n in that it collects different
- * stats; more finds are actually done than the user ever asks for.
- */
+#ifndef TESTING
+int ObjectTable::showStatus(outbuffer_t *out, int verbose) {
 
-static long user_obj_lookups = 0, user_obj_found = 0;
+    std::stringstream ss;
+    if (verbose == 1) {
+        ss <<  "Object name hash table status:" << std::endl;
+        ss <<  "Object name hash table status:" << std::endl;
+        ss << "------------------------------" << std::endl;
+        ss << "Elements:        " << objects_.size() << std::endl;
+        ss << "Memory used:     " << objects_.size() * sizeof(Value) << std::endl;
+        ss << "Bucket count:    " << objects_.bucket_count() << std::endl;
+        ss << "Load factor:     " << objects_.load_factor() << std::endl;
 
-object_t *lookup_object_hash(const char *s) {
-  object_t *ob = find_obj_n(s);
+        outbuf_add(out, ss.str().c_str());
+      }
 
-  user_obj_lookups++;
-  if (ob) {
-    user_obj_found++;
-  }
-  return (ob);
+    if (!verbose) {
+        ss << "Memory used:     " << objects_.size() * sizeof(Value) << std::endl;
+        outbuf_add(out, ss.str().c_str());
+    }
+    return objects_.size() * sizeof(Value);
 }
+#endif
 
-/*
- * Print stats, returns the total size of the object table.  All objects
- * are in table, so their size is included as well.
- */
-
-static char sbuf[100];
-
-int show_otable_status(outbuffer_t *out, int verbose) {
-  const auto object_hash_table_size = CONFIG_INT(__OBJECT_HASH_TABLE_SIZE__);
-
-  int starts;
-
-  if (verbose == 1) {
-    outbuf_add(out, "Object name hash table status:\n");
-    outbuf_add(out, "------------------------------\n");
-    sprintf(sbuf, "%10.2f", objs_in_table / static_cast<float>(object_hash_table_size));
-    outbuf_addv(out, "Average hash chain length:       %s\n", sbuf);
-    sprintf(sbuf, "%10.2f", static_cast<float>(obj_probes) / obj_searches);
-    outbuf_addv(out, "Average search length:           %s\n", sbuf);
-    outbuf_addv(out, "Internal lookups (succeeded):    %lu (%lu)\n",
-                obj_searches - user_obj_lookups, objs_found - user_obj_found);
-    outbuf_addv(out, "External lookups (succeeded):    %lu (%lu)\n", user_obj_lookups,
-                user_obj_found);
-  }
-  starts = static_cast<long>(object_hash_table_size) * sizeof(object_t *) +
-           objs_in_table * sizeof(object_t);
-
-  if (!verbose) {
-    outbuf_addv(out, "Obj table overhead:\t\t%8d %8d\n",
-                object_hash_table_size * sizeof(object_t *), starts);
-  }
-  return starts;
-}
+std::string ObjectTable::basename(std::string s) {
```

Comment:
why does this has to be method of ObjecTable ? It doesn't seems to use any member. Please change this to a standalone utility function. 

Also, fix the var naming, it's almost impossible to understand, at least add some document stating what it should do.

Reply:
This is just the old function rewritten in C++ to be compatible with std::string. The logic is the same.  I will move the function to outside object table. I was just worried about polluting the namespace with needless utility functions.

Reply:
Fixed.